cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -395,98 +398,160 @@ def certificate_activation_handler(request, course_key_string):
     return HttpResponse(status=200)
 
 
-@login_required
-@require_http_methods(("GET", "POST"))
-@ensure_csrf_cookie
-def certificates_list_handler(request, course_key_string):
+class CertificatesListView(View):
     """
     A RESTful handler for Course Certificates
+    """
+
+    permission_classes = [IsAuthenticated]
 
-    GET
+    def get(self, request, course_key_string):
+        """
         html: return Certificates list page (Backbone application)
-    POST
-        json: create new Certificate
-    """
-    course_key = CourseKey.from_string(course_key_string)
-    store = modulestore()
-    with store.bulk_operations(course_key):
-        try:
-            course = _get_course_and_check_access(course_key, request.user)
-        except PermissionDenied:
-            msg = _('PermissionDenied: Failed in authenticating {user}').format(user=request.user)
-            return JsonResponse({"error": msg}, status=403)
-
-        if 'text/html' in request.META.get('HTTP_ACCEPT', 'text/html'):
-            if use_new_certificates_page(course_key):
-                return redirect(get_certificates_url(course_key))
-            certificates_context = get_certificates_context(course, request.user)
-            return render_to_response('certificates.html', certificates_context)
-        elif "application/json" in request.META.get('HTTP_ACCEPT'):
-            # Retrieve the list of certificates for the specified course
-            if request.method == 'GET':
+        """
+        course_key = CourseKey.from_string(course_key_string)
+        store = modulestore()
+
+        with store.bulk_operations(course_key):
+            try:
+                course = _get_course_and_check_access(course_key, request.user)
+            except PermissionDenied:
+                msg = _('PermissionDenied: Failed in authenticating {user}').format(user=request.user)
+                return JsonResponse({"error": msg}, status=403)
+
+            if 'text/html' in request.META.get('HTTP_ACCEPT', 'text/html'):
+                if use_new_certificates_page(course_key):
+                    return redirect(get_certificates_url(course_key))
+                certificates_context = get_certificates_context(course, request.user)
+                return render_to_response('certificates.html', certificates_context)
+
+            elif "application/json" in request.META.get('HTTP_ACCEPT'):
                 certificates = CertificateManager.get_certificates(course)
                 return JsonResponse(certificates, encoder=EdxJSONEncoder)
-            elif request.method == 'POST':
-                # Add a new certificate to the specified course
+
+            else:
+                return HttpResponse(status=406)
+
+    def post(self, request, course_key_string):
```

Comment:
```suggestion
    @method_decorator(ensure_csrf_cookie)
    def post(self, request, course_key_string):
```

This decorator is applied here to match the original API, which ensures the CSRF cookie is set. Please consider applying it to other HTTP methods too.